pkg/usermanagement/client.go

```diff
@@ -798,6 +825,58 @@ func (c *Client) AuthenticateWithCode(ctx context.Context, opts AuthenticateWith
 	return body, err
 }
 
+// AuthenticateWithRefreshToken obtains a new AccessToken and RefreshToken for
+// an existing session
+func (c *Client) AuthenticateWithRefreshToken(ctx context.Context, opts AuthenticateWithRefreshTokenOpts) (RefreshAuthenticationResponse, error) {
+	payload := struct {
+		AuthenticateWithRefreshTokenOpts
+		ClientSecret string `json:"client_secret"`
+		GrantType    string `json:"grant_type"`
+	}{
+		AuthenticateWithRefreshTokenOpts: opts,
+		ClientSecret:                     c.APIKey,
+		GrantType:                        "refresh_token",
+	}
+
+	jsonData, err := json.Marshal(payload)
+	if err != nil {
+		return RefreshAuthenticationResponse{}, err
+	}
+
+	req, err := http.NewRequest(
+		http.MethodPost,
+		c.Endpoint+"/user_management/authenticate",
+		bytes.NewBuffer(jsonData),
+	)
+
+	if err != nil {
+		return RefreshAuthenticationResponse{}, err
+	}
+
+	// Add headers and context to the request
+	req = req.WithContext(ctx)
```

Comment:
I know this is just following existing patterns, and if we refactor it should be done separately, but I wonder why we don't call [`http.NewRequestWithContext`](https://pkg.go.dev/net/http#NewRequestWithContext) initially, rather than separate reassigning `req` later like this.